### ISSUE TITLE
Can't resolve url() in .styl files

### ISSUE THREAD

**Issue by powerpufffs** [ym-project/esbuild-stylus-loader#4]
Hey! Thanks so much for making this plugin 🙌🏿

I've been trying to migrate our react build tools from webpack to esbuild. 

After installing your plugin, I keep running into this error:

```
stylus:/Users/ztai/Code/praxis/praxis_app/client/src/common/styles/global.styl:644:11: error: Could not resolve "/rp/fonts/ibm-plex-mono/IBMPlexMono-Regular.woff2" (the plugin "stylus-loader" didn't set a resolve directory)
stylus:/Users/ztai/Code/praxis/praxis_app/client/src/common/styles/global.styl:644:85: error: Could not resolve "/rp/fonts/ibm-plex-mono/IBMPlexMono-Regular.woff" (the plugin "stylus-loader" didn't set a resolve directory)
stylus:/Users/ztai/Code/praxis/praxis_app/client/src/common/styles/global.styl:650:11: error: Could not resolve "/rp/fonts/ibm-plex-mono/IBMPlexMono-Italic.woff2" (the plugin "stylus-loader" didn't set a resolve directory)
stylus:/Users/ztai/Code/praxis/praxis_app/client/src/common/styles/global.styl:650:84: error: Could not resolve "/rp/fonts/ibm-plex-mono/IBMPlexMono-Italic.woff" (the plugin "stylus-loader" didn't set a resolve directory)
stylus:/Users/ztai/Code/praxis/praxis_app/client/src/common/styles/global.styl:656:11: error: Could not resolve "/rp/fonts/ibm-plex-mono/IBMPlexMono-Bold.woff2" (the plugin "stylus-loader" didn't set a resolve directory)
...
```

The code in question looks like this:
```css
@font-face {
  font-family: 'IBM-Plex-Mono';
  font-style: normal;
  font-weight: 400;
  src: url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Regular.woff2') format('woff2'), url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Regular.woff') format('woff');
}
@font-face {
  font-family: 'IBM-Plex-Mono';
  font-style: italic;
  font-weight: 400;
  src: url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Italic.woff2') format('woff2'), url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Italic.woff') format('woff');
}
@font-face {
  font-family: 'IBM-Plex-Mono';
  font-style: normal;
  font-weight: 500;
  src: url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Bold.woff2') format('woff2'), url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-Bold.woff') format('woff');
}
@font-face {
  font-family: 'IBM-Plex-Mono';
  font-style: italic;
  font-weight: 500;
  src: url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-BoldItalic.woff2') format('woff2'), url($static_host + '/rp/fonts/ibm-plex-mono/IBMPlexMono-BoldItalic.woff') format('woff');
}
```

I'm brand new to both esbuild and webpack so I apologize if this is a dumb mistake. My esbuild method looks like the following:

```javascript
const start = async () => {
  try {
    await build({
      entryPoints: ["./src/adminApp/index.tsx", "./src/onboardingSurvey/index.tsx"],
      //   entryNames: "[dir]",
      outdir: path.resolve(__dirname, "../server/public2"),
      bundle: true,
      sourcemap: true,
      minify: true,
      //   target: ["chrome58", "firefox57", "edge16"],
      target: "esnext",
      loader: {
        ".svg": "dataurl",
        ".png": "dataurl",
        ".woff2": "file",
        ".woff": "file",
        ".otf": "file",
      },
      external: ["d3-array"],
      plugins: [
        stylusLoader({
          define: [["$static_host", process.env.STATIC_HOST || ""]],
          include: [path.join(__dirname, "src/common/styles")],
          import: [path.join(__dirname, "src/common/styles/variables.styl")],
        }),
      ],
    });
  } catch (e) {
    console.error("e", e);
    throw Error(e);
  }
};
```

Here's what webpack looks like (I didn't write this):
```javascript
{
  test: /\.styl$/i,
  use: [
    { loader: "style-loader" },
    {
      loader: "css-loader",
      options: {
        url: false,
        modules: {
          mode: "global",
          localIdentName: "__[local]__[hash:base64:12]",
        },
        esModule: true,
      },
    },
    {
      loader: "stylus-loader",
      options: {
        stylusOptions: {
          define: [["$static_host", process.env.STATIC_HOST || ""]],
          include: [path.join(__dirname, "src/common/styles")],
          import: [path.join(__dirname, "src/common/styles/variables.styl")],
        },
      },
    },
  ],
},
```

I did some digging to see if this problem was also encountered in webpack, and I found these answers:

https://stackoverflow.com/questions/33450953/webpack-stylus-loader-incorrectly-resolve-url-paths
https://github.com/webpack-contrib/stylus-loader/issues/102

I'm not sure how this fits in with your plugin...
Appreciate your work!

**Comment by ym-project** [comment]
Hi! Thank you for your attention.
I have wanted to rewrite this plugin because esbuild went far ahead. But I don't have enough time to do this.
I will try to resolve this problem at the weekend.

**Comment by ym-project** [comment]
Hi again!

I understood why this error is occurred. Esbuild plugin should handle stylus and non-stylus files himself and mark non-stylus files as external.

I think I fixed it. I will publish test version soon and you will can try it.

**Comment by ym-project** [comment]
@powerpufffs hi!
I'm so sorry what I ignored you for several months. Work, moving to a new home, COVID... so I didn't have time and energy to resolve this.

If you are still interested in this plugin, I released a beta version of plugin `esbuild-stylus-loader@beta`.

Beta version has one backwards-incompatible change: plugin options were moved to the `stylusOptions` object.
```
stylusLoader({
    stylusOptions: {
        import: [.....],
        include: [.....],
        // ...
        // etc
    },
})
```
